app/components/post_draft/draft_input/index.tsx

```diff
@@ -89,41 +93,47 @@ export default function DraftInput({
     cursorPosition,
 }: Props) {
     const theme = useTheme();
+    const isTablet = useIsTablet();
 
-    // const [top, setTop] = useState(0);
+    const [top, setTop] = useState(0);
 
-    // const handleLayout = useCallback((e: LayoutChangeEvent) => {
-    //     setTop(e.nativeEvent.layout.y);
-    // }, []);
+    const handleLayout = useCallback((e: LayoutChangeEvent) => {
+        setTop(e.nativeEvent.layout.y);
+    }, []);
 
     // Render
     const postInputTestID = `${testID}.post.input`;
     const quickActionsTestID = `${testID}.quick_actions`;
     const sendActionTestID = `${testID}.send_action`;
     const style = getStyleSheet(theme);
 
+    // From component/navigation_header/index.tsx and screen/channel/channel.tsx
+    const {defaultHeight} = useHeaderHeight(false, true, false);
+    const autocompleteMaxHeight = Math.min(top - defaultHeight - AUTOCOMPLETE_MARGIN, isTablet ? 200 : 1450);
```

Comment:
magic number here.

you should get the device height and also the keyboard height and then do
`const autocompleteMaxHeight = height - (top + (defaultHeight * 2) + AUTOCOMPLETE_MARGIN + keyboardHeight)`

for the keyboard height you can add this
```typescript
useEffect(() => {
        const keyboardEvent = (event: KeyboardEvent) => {
            setKeyboardHeight(event.endCoordinates.height);
        };
        const shown = Keyboard.addListener('keyboardDidShow', keyboardEvent);
        const hidden = Keyboard.addListener('keyboardDidHide', keyboardEvent);

        return () => {
            shown.remove();
            hidden.remove();
        };
    }, []);
```